abis_mapping/plugins/related_site_id_part_of_lookup.py

```diff
@@ -0,0 +1,45 @@
+"""Module providing custom relatedSiteID lookup checks when value of relationshipToRelatedSite is partOf"""
+
+# Third-party
+import attrs
+import frictionless
+import frictionless.checks
+import frictionless.errors
+
+# Typing
+from typing import Iterable
+
+
+@attrs.define(kw_only=True, repr=False)
+class RelatedSiteIDPartOfLookup(frictionless.Check):
+    """Specific check confirming relationshipToRelatedSite is a valid siteID when relationship is `partOf`.
+
+    Attributes:
+        site_ids: All siteIDs provided in the template.
+    """
+
+    Errors = [frictionless.errors.RowConstraintError]
+    site_ids: set[str]
+
+    def validate_row(self, row: frictionless.Row) -> Iterable[frictionless.Error]:
+        """Validate row (every row).
+
+        Args:
+            row: Row of data to be validated.
+
+        Yields:
+            Errors encountered, if any
+        """
+        check = (
+            row["relatedSiteID"] in self.site_ids
+            if row["relatedSiteID"] and row["relationshipToRelatedSite"].lower().replace(" ", "") == "partof"
```

Comment:
intentionally went this way since it is related to validation and not to mapping, i.e. in the same vane we duplicate enums for fields in the schema.  I am concerned with a bad code smell of adding the vocab dependency to a frictionless plugin module. It may not create a complete circular import but it would be one step away. The example provided was for a mapping function, contained within the module. I could move this custom check into the same module as the site data mapper to completely remove that concern for me but it might make it less readable for others.

Reply:
> intentionally went this way since it is related to validation and not to mapping

ok that makes sense :+1: